### PR TITLE
feat(aws-codepipeline-actions): Add Full Clone support for CodeCommit

### LINKED ISSUE #12236: (codepipeline_actions): Need support for Full Clone for CodeCommit action. (OutputArtifactFormat= CODEBUILD_CLONE_REF)
As referenced [here](https://docs.aws.amazon.com/codepipeline/latest/userguide/tutorials-codecommit-gitclone.html) this is a **Feature Request** to allow CodeBuild to access Git metadata in the pipeline build action. 

*Full Clone* is needed in cases where the CodeBuild action needs to access the full Git Repository metadata

### Use Case

In my case I'm trying to run the AWS Copilot CLI to build & deploy from CodeBuild once there is an update to the CodeCommit repo. AWS Copilot CLI needs access to the Git Repository directly. This is a supported case as referenced in the link above, but the ```codepipeline_actions``` modules seems to have exposed this only for BitBucket sources. It would be nice to extend this to other Sources.

### Proposed Solution

My understanding is that exposing the property ```OutputArtifactFormat``` similar to what is done [here](https://github.com/aws/aws-cdk/blob/b96efd8aaa143845b9fe315a9ee1e8398c4d83c2/packages/%40aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts#L120) should be enough, but I don't understand the source code enough as to make a proper PR.

Special care has to be considered when defining Role permissions as there are specific requirements for this option to work. This is clearly referenced [here](https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodeCommit.html#action-reference-CodeCommit-config)

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by DaWyz**
Add `codeBuildCloneOutput` property to the CodeCommit source action.

It automatically adds the `codecommit:GetRepository` permission to the CodeCommitSourceAction role.
It will also add the `codecommit:GitPull` permission to any CodeBuildAction using the artifact from CodeCommitSourceAction as input.

Closes #12236

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/lib/codebuild/build-action.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { Action } from '../action';
 // keep this import separate from other imports to reduce chance for merge conflicts with v2-main
 // eslint-disable-next-line no-duplicate-imports, import/order
 import { Construct } from '@aws-cdk/core';
+import { CodeCommitSourceAction } from '../codecommit/source-action';
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/codebuild/build-action.ts:11]: This should be moved up, just below the `import { BitBucketSourceAction } from '..';` line.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts [hunk 2]
```diff
@@ -155,11 +180,23 @@ export class CodeCommitSourceAction extends Action {
       ],
     }));
 
+    if (this.props.codeBuildCloneOutput === true) {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/codecommit/source-action.ts:183]: Can you actually fold this statement into the above one, considering their Resource is the same? Something like this:

```ts
      actions: [
        'codecommit:GetBranch',
        'codecommit:GetCommit',
        'codecommit:UploadArchive',
        'codecommit:GetUploadArchiveStatus',
        'codecommit:CancelUploadArchive',
        ...this.props.codeBuildCloneOutput ? ['codecommit:GetRepository'] : [],
      ],
```

File: packages/@aws-cdk/aws-codepipeline-actions/test/codecommit/test.codecommit-source-action.ts [hunk 3]
```diff
@@ -269,6 +269,111 @@ export = {
...
+              'Action': [
+                'logs:CreateLogGroup',
+                'logs:CreateLogStream',
+                'logs:PutLogEvents',
+              ],
+            },
+            {},
+            {},
+            {},
+            {},
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/codecommit/test.codecommit-source-action.ts:339]: If you use the helpers we have like `objectLike()` and `arrayWith()`, you can avoid these empty objects. Here's an example of using them: https://github.com/aws/aws-cdk/blob/12868a2ac9c7016e47633045fb0c00be7eeb5092/packages/%40aws-cdk/pipelines/test/cross-environment-infra.test.ts#L33-L48

File: packages/@aws-cdk/aws-codepipeline-actions/test/codecommit/test.codecommit-source-action.ts [hunk 4]
```diff
@@ -269,6 +269,111 @@ export = {
...
+          ],
+        },
+      }));
+
+      expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+        'PolicyDocument': {
+          'Statement': [
+            {},
+            {},
+            {},
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/codecommit/test.codecommit-source-action.ts:359]: Same comment here as above.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great @DaWyz , thanks so much for the contribution! Couple of minor comments below 😊.

**Comment by skinny85** [comment]
Also, please update the [ReadMe file of the package](https://github.com/aws/aws-cdk/blob/12868a2ac9c7016e47633045fb0c00be7eeb5092/packages/@aws-cdk/aws-codepipeline-actions/README.md#aws-codecommit)!

**Comment by DaWyz** [comment]
@skinny85 PR updated. Please, double check the section added in the `README.md` to make sure it's good enough.

Thanks!

**Review by skinny85 — APPROVED** [review]
This is absolutely fantastic work @DaWyz . Thanks so much for the contribution!

**Pull request merged**